File: Army/combat/rogueability.cpp
#include "rogueability.h"

RogueAbility::RogueAbility(Unit* currentUnit) : Ability(currentUnit) {}

RogueAbility::~RogueAbility() {}

void RogueAbility::attack(Unit* victim) {
    victim->takeDamage(currentUnit->getDamage());
}

File: Java/src/com/gymfox/army/Ability/RogueAbility.java
package com.gymfox.army.Ability;

import com.gymfox.army.Units.Unit;

public class RogueAbility extends Ability {
    public RogueAbility(Unit currentUnit) {
        super(currentUnit);
    }

    @Override
    public void attack(Unit victim) throws Unit.UnitIsDeadException {
        victim.takeDamage(getCurrentUnit().getDamage());
    }
}
